src/components/views/read/cvForm/PersonalInfo.tsx

```diff
@@ -0,0 +1,350 @@
+  const inputRef: RefObject<HTMLInputElement> = useRef(null);
+  const handleClick = (): void => {
+    // 👇️ open file input box on click of another element
+    inputRef.current?.click();
+  };
+  const [inputValues, setInputValues] = useState<{ [key: string]: string }[]>(
+    [],
+  );
+  const handleChange = (field: string, value: string): void => {
+    setInputValues((prevInputValues) => {
+      const updatedInputValues = [...prevInputValues];
+      const fieldValue = { [field]: value };
+      const existingFieldIndex = updatedInputValues.findIndex(
+        (inputValue) => Object.keys(inputValue)[0] === field,
+      );
+
+      if (existingFieldIndex !== -1) {
+        // Field already exists, update its value
+        updatedInputValues[existingFieldIndex] = fieldValue;
+      } else {
+        // Field doesn't exist, add it to the input values
+        updatedInputValues.push(fieldValue);
+      }
+
+      return updatedInputValues;
+    });
+  };
+  const handleFileChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
+    const { files } = event.target;
+    if (files && files.length > 0) {
+      const selectedFile = files[0];
+      const reader = new FileReader();
+      reader.onload = () => {
+        const filePreview = reader.result as string;
+        handleChange('preview', filePreview);
+      };
+      reader.readAsDataURL(selectedFile);
+    }
+  };
+
+  const handleNext = (): void => {
+    const modifiedValues = {
+      'First Name':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'firstName',
+        )?.firstName || '',
+      'Last Name':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'lastName',
+        )?.lastName || '',
+      'Birth Date':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'birthDate',
+        )?.birthDate || '',
+      Gender:
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'gender',
+        )?.gender || '',
+      'Email Address':
+        inputValues.find((inputValue) => Object.keys(inputValue)[0] === 'email')
+          ?.email || '',
+      'Phone Number':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'phoneNumber',
+        )?.phoneNumber || '',
+      Address:
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'address',
+        )?.address || '',
+      'Profile Links':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'profileLinks',
+        )?.profileLinks || '',
+      'Personal Link':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'personalLink',
+        )?.personalLink || '',
+      'Personal Picture':
+        inputValues.find(
+          (inputValue) => Object.keys(inputValue)[0] === 'preview',
+        )?.preview || '',
+    };
```

Comment:
This looks like you are modifying the keys of your object for the current step.

Multiple issues:
- when you go back, even if the values are saved, they are not shown to the user because they now live on a different key than the one you search for to display the value in the inputs.
- this is super long and really inefficient, if you *really need* to map keys from a set to a different one, use something like: 
```ts
const mapping = [{key: 'firstName', label: 'First Name'}, {...}];
```
and you can iterate on this mapping to either show the label instead of the "raw label" or rename your obejct (ut i do not see why you would rename them).